### PR TITLE
Improve documents

### PR DESCRIPTION
**Pull request by ryysud**
**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

Nothing.

**Description of change**

I improve the documents related to spire-server/spire-agent commands and some plugins. 

The changes are below.

- Add missing plugins to Built-in plugins list
- Add missing spire-server's commands and options
- Add missing spire-agent's commands and options
- Add missing sample configurations to each server and agent plugins
- Add IAM policy examples NodeAttestor(server) "aws_iid" and NodeResolver "aws_iid"
- Fix typo
- Remove duplicate word "the"

**Which issue this PR fixes**

Maybe nothing...

### CODE REVIEW

File: doc/spire_agent.md [hunk 1]
```diff
@@ -105,6 +107,16 @@ the following flags are available:
...
+| `-dataDir` | A directory the agent can use for its runtime data | |
+| `-joinToken` | An optional token which has been generated by the SPIRE server | |
+| `-logFile` | File to write logs to | |
+| `-logLevel` | DEBUG, INFO, WARN or ERROR | |
+| `-serverAddress` | IP address or DNS name of the SPIRE server | |
+| `-serverPort` | Port number of the SPIRE server | |
+| `-socketPath` | Location to bind the workload API socket | |
+| `-trustBundle` | Path to the SPIRE server CA bundle | |
+| `-trustDomain` | The trust domain that this agent belongs to | |
+| `-umask` | Umask value to use for new files | |
```
Comment (ryysud) [inline doc/spire_agent.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-agent \
    spire-agent:latest \
    run --help
```

File: doc/spire_agent.md [hunk 2]
```diff
@@ -114,25 +126,30 @@ Calls the workload API to fetch an X509-SVID. This command is aliased to `spire-
 | ---------------- | --------------------------- | ----------------------- |
 | `-silent` | Suppress stdout | |
 | `-socketPath` | Path to the workload API socket | /tmp/agent.sock |
+| `-timeout` | Time to wait for a response | 1s |
 | `-write` | Write SVID data to the specified path | |
```
Comment (ryysud) [inline doc/spire_agent.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-agent \
    spire-agent:latest \
    api fetch --help
```

File: doc/spire_agent.md [hunk 3]
```diff
@@ -114,25 +126,30 @@ Calls the workload API to fetch an X509-SVID. This command is aliased to `spire-
...
+Calls the workload API to fetch a x.509-SVID.
 
 | Command          | Action                      | Default                 |
 | ---------------- | --------------------------- | ----------------------- |
-| `-audience` | A comma separated list of audience values | |
+| `-silent` | Suppress stdout | |
 | `-socketPath` | Path to the workload API socket | /tmp/agent.sock |
-| `-spiffeID` | The SPIFFE ID of the JWT being requested (optional) | |
+| `-timeout` | Time to wait for a response | 1s |
+| `-write` | Write SVID data to the specified path | |
```
Comment (ryysud) [inline doc/spire_agent.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-agent \
    spire-agent:latest \
    api fetch x509 --help
```

File: doc/spire_agent.md [hunk 4]
```diff
@@ -143,6 +160,25 @@ Calls the workload API to validate the supplied JWT-SVID.
...
 | `-svid` | The JWT-SVID to be validated | |
+| `-timeout` | Time to wait for a response | 1s |
+
+### `spire-agent api watch`
+
+Attaches to the workload API and watches for X509-SVID updates, printing details when updates are received.
+
+| Command          | Action                      | Default                 |
+| ---------------- | --------------------------- | ----------------------- |
+| `-socketPath` | Path to the workload API socket | /tmp/agent.sock |
```
Comment (ryysud) [inline doc/spire_agent.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-agent \
    spire-agent:latest \
    api watch --help
```

File: doc/spire_agent.md [hunk 5]
```diff
@@ -143,6 +160,25 @@ Calls the workload API to validate the supplied JWT-SVID.
...
+
+### `spire-agent healthcheck`
+
+Checks SPIRE agent's health.
+
+| Command       | Action                                                             | Default        |
+|:--------------|:-------------------------------------------------------------------|:---------------|
+| `-shallow` | Perform a less stringent health check | |
+| `-socketPath` | Path to the workload API socket | /tmp/agent.sock |
+| `-verbose` | Print verbose information | |
```
Comment (ryysud) [inline doc/spire_agent.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-agent \
    spire-agent:latest \
    healthcheck --help
```

File: doc/spire_server.md [hunk 6]
```diff
@@ -115,7 +116,16 @@ Most of the configuration file above options have identical command-line counter
...
+| `-bindAddress` | IP address or DNS name of the SPIRE server | |
 | `-config` | Path to a SPIRE config file | conf/server/server.conf |
+| `-dataDir` | Directory to store runtime data to | |
+| `-logFile` | File to write logs to | |
+| `-logLevel` | DEBUG, INFO, WARN or ERROR | |
+| `-registrationUDSPath` | UDS Path to bind registration API | |
+| `-serverPort` | Port number of the SPIRE server | |
+| `-trustDomain` | The trust domain that this server belongs to | |
+| `-umask` | Umask value to use for new files | |
+| `-upstreamBundle` | Include upstream CA certificates in the bundle | |
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    run --help
```

File: doc/spire_server.md [hunk 7]
```diff
@@ -135,28 +145,37 @@ Creates registration entries.
...
+| `-entryExpiry`   | An expiry, from epoch in seconds, for the resulting registration entry to be pruned | |
+| `-federatesWith` | A list of trust domain SPIFFE IDs representing the trust domains this registration entry federates with. A bundle for that trust domain must already exist | |
+| `-node`          | If set, this entry will be applied to matching nodes rather than workloads | |
 | `-parentID`      | The SPIFFE ID of this record's parent.                                 |                |
 | `-registrationUDSPath` | Path to the SPIRE server registration api socket | /tmp/spire-registration.sock |
 | `-selector`      | A colon-delimited type:value selector used for attestation. This parameter can be used more than once, to specify multiple selectors that must be satisfied. | |
 | `-spiffeID`      | The SPIFFE ID that this record represents and will be set to the SVID issued. | |
 | `-ttl`           | A TTL, in seconds, for any SVID issued as a result of this record.     | 3600           |
-| `-federatesWith` | A list of trust domain SPIFFE IDs representing the trust domains this registration entry federates with. A bundle for that trust domain must already exist | |
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    entry create --help
```

File: doc/spire_server.md [hunk 8]
```diff
@@ -135,28 +145,37 @@ Creates registration entries.
...
+| `-entryID`       | The Registration Entry ID of the record to update                      |                |
+| `-federatesWith` | A list of trust domain SPIFFE IDs representing the trust domains this registration entry federates with. A bundle for that trust domain must already exist | |
 | `-parentID`      | The SPIFFE ID of this record's parent.                                 |                |
 | `-registrationUDSPath` | Path to the SPIRE server registration api socket | /tmp/spire-registration.sock |
 | `-selector`      | A colon-delimited type:value selector used for attestation. This parameter can be used more than once, to specify multiple selectors that must be satisfied. | |
 | `-spiffeID`      | The SPIFFE ID that this record represents and will be set to the SVID issued. | |
 | `-ttl`           | A TTL, in seconds, for any SVID issued as a result of this record.     | 3600           |
-| `-federatesWith` | A list of trust domain SPIFFE IDs representing the trust domains this registration entry federates with. A bundle for that trust domain must already exist | |
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    entry update --help
```

File: doc/spire_server.md [hunk 9]
```diff
@@ -173,7 +192,9 @@ Displays configured registration entries.
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    entry show --help
```

File: doc/spire_server.md [hunk 10]
```diff
@@ -216,6 +237,33 @@ Deletes bundle data for a trust domain. This command cannot be used to delete th
...
 | `-registrationUDSPath` | Path to the SPIRE server registration api socket | /tmp/spire-registration.sock |
 
+### `spire-server agent evict`
+
+De-attesting an already attested node given its spiffeID.
+
+| Command       | Action                                                             | Default        |
+|:--------------|:-------------------------------------------------------------------|:---------------|
+| `-registrationUDSPath` | Path to the SPIRE server registration api socket | /tmp/spire-registration.sock |
+| `-spiffeID` | The SPIFFE ID of the agent to evict (agent identity) | |
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    agent evict --help
```

File: doc/spire_server.md [hunk 11]
```diff
@@ -216,6 +237,33 @@ Deletes bundle data for a trust domain. This command cannot be used to delete th
...
+
+### `spire-server agent list`
+
+Displays attested nodes.
+
+| Command       | Action                                                             | Default        |
+|:--------------|:-------------------------------------------------------------------|:---------------|
+| `-registrationUDSPath` | Path to the SPIRE server registration api socket | /tmp/spire-registration.sock |
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    agent list --help
```

File: doc/spire_server.md [hunk 12]
```diff
@@ -216,6 +237,33 @@ Deletes bundle data for a trust domain. This command cannot be used to delete th
...
+
+### `spire-server healthcheck`
+
+Checks SPIRE server's health.
+
+| Command       | Action                                                             | Default        |
+|:--------------|:-------------------------------------------------------------------|:---------------|
+| `-registrationUDSPath` | Path to the SPIRE server registration api socket | /tmp/spire-registration.sock |
+| `-shallow` | Perform a less stringent health check | |
+| `-verbose` | Print verbose information | |
```
Comment (ryysud) [inline doc/spire_server.md]:
```
docker run --rm \
    --entrypoint /opt/spire/bin/spire-server \
    spire-server:latest \
    healthcheck --help
```

**Comment by ajessup** [comment]
Many thanks for this! Great to have the documentation updated, and the config snippets are really helpful.

**Comment by ryysud** [comment]
I force-pushed commits to this PR to pass the DCO.

File: doc/spire_agent.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryysud) [reply doc/spire_agent.md]: I added the `umask` without understanding that it is intended to be undocumented command.

Fix: https://github.com/spiffe/spire/pull/873/commits/1d68c4d14c103037aca0fe6515219762f9c53378

File: doc/spire_server.md [hunk 13]
```diff
@@ -115,7 +116,16 @@ Most of the configuration file above options have identical command-line counter
...
 |:-----------------|:----------------------------|:------------------------|
+| `-bindAddress` | IP address or DNS name of the SPIRE server | |
 | `-config` | Path to a SPIRE config file | conf/server/server.conf |
+| `-dataDir` | Directory to store runtime data to | |
+| `-logFile` | File to write logs to | |
+| `-logLevel` | DEBUG, INFO, WARN or ERROR | |
+| `-registrationUDSPath` | UDS Path to bind registration API | |
+| `-serverPort` | Port number of the SPIRE server | |
+| `-trustDomain` | The trust domain that this server belongs to | |
+| `-umask` | Umask value to use for new files | |
```
[the remark this answers is not in the document]
Comment (ryysud) [reply doc/spire_server.md]: Fix: https://github.com/spiffe/spire/pull/873/commits/fa4b3a26dd4f08af8b987a50d75d8a5a669eca4b

**Comment by ryysud** [comment]
When the review is complete, I will rebase all commits like `fixup: xxx`.

**Comment by ryysud** [comment]
Thanks for your review 😄

**Pull request merged**